api/jobset/v1alpha2/jobset_types.go

```diff
@@ -169,7 +169,7 @@ type JobSetStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
 	// Restarts tracks the number of times the JobSet has restarted (i.e. recreated in case of RecreateAll policy).
-	Restarts int32 `json:"restarts,omitempty"`
+	Restarts int32 `json:"restarts"`
```

Comment:
Add optional to this
```suggestion
	// +optional
	Restarts int32 `json:"restarts"`
```
This will keep the field as optional, but mean that a go client would render the zero value. This means structured clients will always set an opinion "if it wasn't zero but I'm now updating the object, it is now zero", which I _think_ is what you want

Unstructured clients would still treat this as optional and their behaviour wouldn't change

Round trips through both sets of clients are preserved, apart from on the initial write of any structured go based client which will set the zero value

Reply:
Thanks. I added optional in the comment. IIUC, adding pointer to the type will still be a breaking change, so I don't make further edits here.